### PR TITLE
Fix racing condition using leasing on blob

### LINKED ISSUE #22: Racing condition when multiple masters are creating keys at the same time

### PR DESCRIPTION
**Pull request by ritazh**
fixes #22

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -125,8 +154,61 @@ func getVault(subscriptionID string, vaultName string) (vaultUrl *string, err er
 	return vault.Properties.VaultURI, nil
 }
 
-func createKey(keyClient kv.ManagementClient, vaultUrl string, keyName string) (*string, error) {
+func createKey(keyClient kv.ManagementClient, vaultUrl string, keyName string, providerVaultName string, subscriptionID string) (*string, error) {
 	fmt.Println("Key not found. Creating a new key...")
+	storageAccountsClient := storagemgmt.NewAccountsClient(subscriptionID)
```
Comment (khenidak) [inline server.go]: We really don't subscription id here. You can get a token for resource by storage account name, and env (cloud name) only. This way we ask the user for less

File: server.go [hunk 2]
```diff
@@ -125,8 +154,61 @@ func getVault(subscriptionID string, vaultName string) (vaultUrl *string, err er
...
+	token, _ := GetManagementToken(AuthGrantType(), configFilePath)
+	storageAccountsClient.Authorizer = token
+
+	resourceGroup, err := GetResourceGroup(configFilePath)
+	if err != nil {
+		return nil, err
+	}
+	storageAcctName := providerVaultName
+	res, err := storageAccountsClient.ListKeys(*resourceGroup, storageAcctName)
```
Comment (khenidak) [inline server.go]: You also don't need that.

**Comment by khenidak** [comment]
Shouldn't the configuration file carry the storage account name used for leases?

File: server.go (hunk 1 above)
Comment (ritazh) [reply server.go]: Also doesnt seem like the Azure Go SDK has the AD authentication feature added for storage yet: https://github.com/Azure/azure-sdk-for-go/blob/v18.0.0/storage/client.go#L269-L276

```
func NewBasicClient(accountName, accountKey string) (Client, error) {
	if accountName == StorageEmulatorAccountName {
		return NewEmulatorClient()
	}
	return NewClient(accountName, accountKey, DefaultBaseURL, DefaultAPIVersion, defaultUseHTTPS)
}
```

**Comment by ritazh** [comment]
> Shouldn't the configuration file carry the storage account name used for leases?

Instead of adding yet another field in the configuration file, I'm using the same name as the keyvault for the storage account.

**Comment by khenidak** [comment]
Let us have `use AAD auth as future feature (issue?)` and merge this as is
/lgtm

**Comment by ritazh** [comment]
Tracking issue here: https://github.com/Azure/kubernetes-kms/issues/26

**Pull request merged**